commons-geometry-euclidean/src/main/java/org/apache/commons/geometry/euclidean/twod/hull/ConvexHull2D.java

```diff
@@ -0,0 +1,469 @@
+
+                if (point.equals(v2)) {
+                    return true;
+                }
+
+                // do side of line test: multiply the last location with this location
+                // if they are the same sign then the operation will yield a positive result
+                // -x * -y = +xy, x * y = +xy, -x * y = -xy, x * -y = -xy
+                if (last * signedAreaPoints(v1, v2, point) < 0) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /** Compute the signed area of the parallelogram formed by vectors between the given points. The first
+         * vector points from {@code p0} to {@code p1} and the second from {@code p0} to {@code p3}.
+         * @param p0 first point
+         * @param p1 second point
+         * @param p2 third point
+         * @return signed area of parallelogram formed by vectors between the given points
+         */
+        private static double signedAreaPoints(final Vector2D p0, final Vector2D p1, final Vector2D p2) {
+            return p0.vectorTo(p1).signedArea(p0.vectorTo(p2));
+        }
+
+        /**
+         * Find the convex hull vertices from the set of input points.
+         * @param points the set of input points
+         * @return the convex hull vertices in CCW winding
+         */
+        private Collection<Vector2D> findHullVertices(final Collection<Vector2D> points) {
+
+            final List<Vector2D> pointsSortedByXAxis = new ArrayList<>(points);
+
+            // sort the points in increasing order on the x-axis
+            pointsSortedByXAxis.sort((o1, o2) -> {
+                // need to take the tolerance value into account, otherwise collinear points
+                // will not be handled correctly when building the upper/lower hull
+                final int cmp = precision.compare(o1.getX(), o2.getX());
+                if (cmp == 0) {
+                    return precision.compare(o1.getY(), o2.getY());
+                } else {
+                    return cmp;
+                }
+            });
+
+            // build lower hull
+            final List<Vector2D> lowerHull = new ArrayList<>();
+            for (final Vector2D p : pointsSortedByXAxis) {
+                updateHull(p, lowerHull);
+            }
+
+            // build upper hull
+            final List<Vector2D> upperHull = new ArrayList<>();
+            for (int idx = pointsSortedByXAxis.size() - 1; idx >= 0; idx--) {
+                final Vector2D p = pointsSortedByXAxis.get(idx);
+                updateHull(p, upperHull);
+            }
+
+            // concatenate the lower and upper hulls
+            // the last point of each list is omitted as it is repeated at the beginning of the other list
+            final List<Vector2D> hullVertices = new ArrayList<>(lowerHull.size() + upperHull.size() - 2);
+            for (int idx = 0; idx < lowerHull.size() - 1; idx++) {
```

Comment:
Can these loops be changed to omit the first point of each half:
```java
for (int idx = 1; idx < lowerHull.size(); idx++) {
```

Reply:
I am going to check that tomorrow .

Reply:
Changing which point to omit was no problem.